### ISSUE TITLE
Loss Spikes and Does Not Converge (MSE Loss)

### ISSUE THREAD

**Issue by Wendysigh** [chao1224/NeuralMD#2]
Thank you for providing this codebase—it has been very helpful! However, I have encountered an issue where the training loss (MSE) spikes and does not converge when running the code.

I initially tested the code with my own dataset, but the loss remained unstable and did not decrease over epochs. 
To ensure it wasn’t an issue with my dataset, I used the MISATO100 dataset along with the best parameter settings mentioned in the README. Unfortunately, the loss still fails to converge and exhibits spiking behavior.

Is there any specific preprocessing or configuration required beyond what is provided in the documentation? Any help or guidance would be greatly appreciated!

Thank you!

**Comment by chao1224** [comment]
Hi @Wendysigh,

There are some tricks in the hyperparameter. My server is down for maintenance, so I can't find the log files for now. But I remembered it is related to the initial velocity modeling. Can you try adding `--use_MLP_velocity` in your argument?

**Comment by Wendysigh** [comment]
Hi @chao1224 
Thanks for the prompt reply. I tried adding  --use_MLP_velocity in the argument. The parameter for MISATO 100 looks like:

```python examples/main_MISATO_multi_traj_NeuralMD.py --seed=22 --lr=1e-4 --optimizer=Adam \
    --lr_scheduler=None --FrameNet_num_radial=100 --num_workers=0 --epochs=100 --batch_size=8 --dataset=MISATO_100 \
     --NeuralMD_step_size=5 --NeuralMD_scaling=100 --NeuralMD_velocity_refined_value_coefficient=0 \
     --ODE_method=euler --NeuralMD_binding_model=NeuralMD_Binding01 --no_MLP_velocity \
     --no_NeuralMD_Binding_start_with_first_frame --NeuralMD_Binding_frame_num=20 --wandb --use_MLP_velocity
```
but the loss spike still exists. I wonder if there any other important parameters?

**Comment by chao1224** [comment]
Hi @Wendysigh ,

If you use `use_MLP_velocity `, then it should be very close to what we reported in the paper. Have you tried the single-trajectory case?

Here are three more insights:
- We initially focused on the single-trajectory case, where the generalization process performed as expected. However, when we extended our experiments to the multi-trajectory case, we observed that generalization became more challenging (fluctuation of the training loss).
- An insight from the computer vision domain is that even if the loss of ODE/SDE flow models does not decrease, their performance metrics (e.g., FID) can still show improvement.
- Following this observation, we report the ones with better generalization performance on the validation set (for multi-traj).
